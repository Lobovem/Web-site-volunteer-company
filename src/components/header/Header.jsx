import layout from '../../styles/layout.module.scss';
import { LangSite } from '../LangSite/LangSite';
import { Menu } from '../Menu/Menu';
import { LogoMain } from '../LogoMain/LogoMain';
import { Burger } from '../kit/Burger/Burger';
import s from './header.module.scss';

export const Header = () => {
  return (
    <header className={`${s.header} ${layout.layout}`}>
      <div className={s.header__wrap}>
        <LogoMain></LogoMain>
        <div className={s.header__menuWrap}>
          <Burger></Burger>
          <Menu></Menu>
          <LangSite href="/"></LangSite>
        </div>
      </div>
    </header>
  );
};
