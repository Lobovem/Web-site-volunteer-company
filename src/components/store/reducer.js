import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  burgerMenuState: false,
};

export const appReducer = createSlice({
  name: 'app',
  initialState: initialState,
  reducers: {
    isOpenBurgerMenu: (state, action) => {
      state.burgerMenuState = !state.burgerMenuState;
    },
  },
});

export const burgerMenuSelector = (state) => state.app.burgerMenuState;

export const { isOpenBurgerMenu } = appReducer.actions;
export default appReducer.reducer;
