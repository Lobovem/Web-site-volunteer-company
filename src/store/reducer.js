import { createSlice } from '@reduxjs/toolkit';

const initialState = {
  todoListState: [],
};

export const appReducer = createSlice({
  name: 'todoList',
  initialState: initialState,
  reducers: {
    // add_todo: (state, action) => {
    //   state.todoListState = [
    //     ...state.todoListState,
    //     {
    //       id: Math.random(),
    //       task: action.payload,
    //       complete: false,
    //       isEditing: false,
    //     },
    //   ];
    // },
    // delete_todo: (state, action) => {
    //   state.todoListState = state.todoListState.filter((todo) => todo.id !== action.payload.id);
    // },
  },
});

// export const userInputSelector = (state) => state.todoList.inputState;
// export const todoListSelector = (state) => state.todoList.todoListState;

// export const { add_todo } = todoReducer.actions;
export default appReducer.reducer;
