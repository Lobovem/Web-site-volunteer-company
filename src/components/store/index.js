// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { appReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: appReducer.reducer,
  },
});
