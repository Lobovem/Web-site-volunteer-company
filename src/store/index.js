// import configureStore from '@reduxjs/toolkit';
import { configureStore } from '@reduxjs/toolkit';
import { todoReducer } from './reducer';

export const store = configureStore({
  reducer: {
    todoList: todoReducer.reducer,
  },
});
