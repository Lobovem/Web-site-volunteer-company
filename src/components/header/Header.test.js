import { render, screen } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import { Header } from './Header';
import { Provider } from 'react-redux';
import { store } from '../../redux/config/store';

describe('Header` component', () => {
  test('renders Header component', () => {
    // eslint-disable-next-line testing-library/render-result-naming-convention
    const component = render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    expect(component).toMatchSnapshot();
  });

  test('renders and search elements links', () => {
    render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    const phoneLinks = screen.getAllByRole('link'); // Search all links
    expect(phoneLinks).toHaveLength(3);
  });

  test('renders and search Header component with 1 images', () => {
    render(
      <Provider store={store}>
        <Router>
          <Header />
        </Router>
      </Provider>
    );
    const phoneLinks = screen.getAllByRole('img'); // Search all links
    expect(phoneLinks).toHaveLength(1);
  });
});
