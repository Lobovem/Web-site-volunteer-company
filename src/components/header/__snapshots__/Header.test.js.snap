// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header\` component renders Header component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="header"
      >
        <div
          class="header__wrap"
        >
          <a
            class="logoMain"
            href="/"
          >
            <img
              alt="ukraine-flag.png"
              src="./img/logo-main.svg"
            />
          </a>
          <div
            class="header__menuWrap"
          >
            <div
              class="burger"
              data-testid="burger"
            >
              <span
                class="burger__line"
                data-testid="burger-line"
              />
              <span
                class="burger__line"
                data-testid="burger-line"
              />
              <span
                class="burger__line"
                data-testid="burger-line"
              />
            </div>
            <nav
              class="menu"
            >
              <ul
                class="menu__list"
              />
            </nav>
            <div
              class="langWrap"
            >
              <a
                class="langWrap__link langWrap__link_active"
                href="#"
              >
                UA
              </a>
              <a
                class="langWrap__link"
                href="#"
              >
                EN
              </a>
            </div>
          </div>
        </div>
      </header>
    </div>
  </body>,
  "container": <div>
    <header
      class="header"
    >
      <div
        class="header__wrap"
      >
        <a
          class="logoMain"
          href="/"
        >
          <img
            alt="ukraine-flag.png"
            src="./img/logo-main.svg"
          />
        </a>
        <div
          class="header__menuWrap"
        >
          <div
            class="burger"
            data-testid="burger"
          >
            <span
              class="burger__line"
              data-testid="burger-line"
            />
            <span
              class="burger__line"
              data-testid="burger-line"
            />
            <span
              class="burger__line"
              data-testid="burger-line"
            />
          </div>
          <nav
            class="menu"
          >
            <ul
              class="menu__list"
            />
          </nav>
          <div
            class="langWrap"
          >
            <a
              class="langWrap__link langWrap__link_active"
              href="#"
            >
              UA
            </a>
            <a
              class="langWrap__link"
              href="#"
            >
              EN
            </a>
          </div>
        </div>
      </div>
    </header>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
